File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 8/13/2025, 8:23:59 PM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int a=0;
       int max_sum=nums[0];
       for(int x:nums){
           a=a+x;
           max_sum=max(max_sum,a);
           a=max(a,0);
       }
        return max_sum;
    }
};

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 8/13/2025, 8:23:57 PM
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int maxSum = Integer.MIN_VALUE;

        for(int i=0;i<nums.length;i++){
            if(sum<0){
                sum = nums[i];
            }
            else{
                sum = sum + nums[i];
            }
            maxSum = Math.max(maxSum , sum);
        }
        return maxSum;
    }
}